pallets/xvm/src/mock.rs

```diff
@@ -131,12 +132,22 @@ impl astar_primitives::ethereum_checked::AccountMapping<AccountId> for HashedAcc
     }
 }
 
+thread_local! {
+    static TRANSACTED: RefCell<Option<(H160, CheckedEthereumTx)>> = RefCell::new(None);
+}
+
 pub struct MockEthereumTransact;
+impl MockEthereumTransact {
+    pub(crate) fn assert_transacted(source: H160, checked_tx: CheckedEthereumTx) {
+        assert!(TRANSACTED.with(|v| v.eq(&RefCell::new(Some((source, checked_tx))))));
```

Comment:
Is the approach with `RefCell` better than just borrowing the value and comparing directly to `Some(...)`?

Asking since I don't know 🙂

Reply:
Both of them work, I just used this one randomly as it's just a mock. :]

Inspired by your "easier for debug" comment above, I'll change this to `assert_eq!` instead.